StripePaymentsUI/StripePaymentsUI/Source/Internal/UI/Views/STPFormTextField.swift

```diff
@@ -197,7 +197,17 @@ import UIKit
     /// :nodoc:
     @objc
     public override func deleteBackward() {
-        super.deleteBackward()
+        // `UITextField.deleteBackwards` doesn't update the `text` property directly, and we depend on the `didSet`
+        // call on it to update our backing store.
+        // To get around this we manually remove the last character if the text is not empty, instead of forwarding
+        // the `deleteBackwards` call.
+        if var text = text, text.count > 0 {
+            text.removeLast()
+            self.text = text
+        } else {
+            super.deleteBackward()
```

Comment:
Why isn't forwarding this an issue when it's not the last character? Seems like we should always manually remove the character.

Reply:
This method is kinda weird, it only gets called when you delete on an empty field, but we also call it directly [here](https://github.com/stripe/stripe-ios/blob/d8e8b4bbdea990a4c1514672a4e3df4137457c03/StripePaymentsUI/StripePaymentsUI/Source/UI%20Components/STPPaymentCardTextField.swift#L1608), which is where the alternative behavior is triggered.

This actually makes me think that it might be better to have this alternate logic at the caller instead of here